pandas/tests/strings/test_strings.py

```diff
@@ -535,6 +535,26 @@ def test_strip_lstrip_rstrip_args(any_string_dtype):
     tm.assert_series_equal(result, expected)
 
 
+def test_remove_suffix_prefix(any_string_dtype):
```

Comment:
Can we parameterize these tests cases using https://docs.pytest.org/en/6.2.x/parametrize.html

Reply:
You mean like this?

```py
@pytest.mark.parametrize(
    "prefix,expected",
    [("x", ["xABCxx", "x BNSD", "LDFJH xx"]), ("xx ", ["xxABCxx", "BNSD", "LDFJH xx"])],
)
def test_removeprefix(any_string_dtype, prefix, expected):
    ser = Series(["xxABCxx", "xx BNSD", "LDFJH xx"], dtype=any_string_dtype)

    result = ser.str.removeprefix(prefix)
    ser_expected = Series(expected, dtype=any_string_dtype)
    tm.assert_series_equal(result, ser_expected)


@pytest.mark.parametrize(
    "suffix,expected",
    [("x", ["xxABCx", "xx BNSD", "LDFJH x"]), ("xx ", ["xxABCxx", "xx BNSD", "LDFJH"])],
)
def test_removesuffix(any_string_dtype, suffix, expected):
    ser = Series(["xxABCxx", "xx BNSD", "LDFJH xx"], dtype=any_string_dtype)

    result = ser.str.removesuffix(suffix)
    ser_expected = Series(expected, dtype=any_string_dtype)
    tm.assert_series_equal(result, ser_expected)
```

It's actually longer and I find it harder to read. But your call?

Reply:
Not quite sure why the strings need to be so complex. A minimal case is surely isomorphic..
```python

@pytest.mark.parametrize(
    "prefix, expected", [("a", ["b", " b c", "bc"]), ("ab", ["", "a b c", "bc"])],
)
def test_removeprefix(any_string_dtype, prefix, expected):
    ser = Series(["ab", "a b c", "bc], dtype=any_string_dtype)
    result = ser.str.removeprefix(prefix)
    ser_expected = Series(expected, dtype=any_string_dtype)
    tm.assert_series_equal(result, ser_expected)

@pytest.mark.parametrize(
    "suffix, expected", [("c", ["ab", "a b ", "b"]), ("bc", ["ab", "a b c", ""])],
)
def test_removesuffix(any_string_dtype, prefix, expected):
    ser = Series(["ab", "a b c", "bc"], dtype=any_string_dtype)
    result = ser.str.removesuffix(suffix)
    ser_expected = Series(expected, dtype=any_string_dtype)
    tm.assert_series_equal(result, ser_expected)
```

Reply:
> Not quite sure why the strings need to be so complex.

Don't know either. I just copied those strings from the other test cases.

Reply:
Lets simplify them then - to what @attack68 suggested above.

Reply:
Yes, already have. Just waiting for feedback on return value type hints and perf benchmark before pushing.